features/lwipstack/LWIPStack.cpp

```diff
@@ -230,14 +230,26 @@ nsapi_error_t LWIP::socket_open(nsapi_socket_t *handle, nsapi_protocol_t proto)
         return NSAPI_ERROR_NO_SOCKET;
     }
 
-    enum netconn_type lwip_proto = proto == NSAPI_TCP ? NETCONN_TCP : NETCONN_UDP;
+    enum netconn_type netconntype;
+    if (proto == NSAPI_TCP) {
+        netconntype = NETCONN_TCP;
+    } else if (proto == NSAPI_UDP) {
+        netconntype = NETCONN_UDP;
+    } else {
+        netconntype = NETCONN_RAW;
```

Comment:
For future-proofing, can you make this error for unknown `proto`? I know it didn't before, but it should have,